components/org.wso2.carbon.identity.workflow.engine/src/main/java/org/wso2/carbon/identity/workflow/engine/ApprovalTaskServiceImpl.java

```diff
@@ -0,0 +1,631 @@
+        } catch (IdentityRoleManagementException e) {
+            throw new WorkflowEngineException(
+                    WorkflowEngineConstants.ErrorMessages.ERROR_OCCURRED_WHILE_RETRIEVING_APPROVAL_TASKS_FOR_USER
+                            .getCode(),
+                    WorkflowEngineConstants.ErrorMessages.ERROR_OCCURRED_WHILE_RETRIEVING_APPROVAL_TASKS_FOR_USER.
+                            getDescription());
+        }
+
+    }
+
+    private long getCreatedTime() {
+
+        return this.localCreatedTime;
+    }
+
+    private void setCreatedTime(long param) {
+
+        this.localCreatedTime = param;
+    }
+
+    private WorkflowRequest getWorkflowRequest(String requestId) throws WorkflowEngineException {
+
+        try {
+            return WorkflowEngineServiceDataHolder.getInstance().getWorkflowManagementService()
+                    .getWorkflowRequest(requestId);
+        } catch (WorkflowException e) {
+            throw new WorkflowEngineException(
+                    WorkflowEngineConstants.ErrorMessages.ERROR_OCCURRED_WHILE_RETRIEVING_WORKFLOW_REQUEST.getCode(),
+                    WorkflowEngineConstants.ErrorMessages.ERROR_OCCURRED_WHILE_RETRIEVING_WORKFLOW_REQUEST.
+                            getDescription());
+        }
+    }
+
+    private void updateStepDetailsOfRequest(String approvalTaskId) throws WorkflowEngineException {
+
+        String workflowRequestId = approvalTaskDAO.getWorkflowRequestIdByApprovalTaskId(approvalTaskId);
+        WorkflowRequest workflowRequest = getWorkflowRequest(workflowRequestId);
+        List<Parameter> parameterList = getParameterList(workflowRequest);
+        String workflowId = approvalTaskDAO.getWorkflowID(approvalTaskId);
+        approvalTaskDAO.deleteApprovalTasksOfWorkflowRequest(workflowRequestId);
+
+        int stepValue = approvalTaskDAO.getCurrentApprovalStepOfWorkflowRequest(workflowRequestId, workflowId);
+        if (stepValue < numOfStates(workflowRequest)) {
+            addApprovalTasksForWorkflowRequest(workflowRequest, parameterList);
+        } else {
+            completeWorkflowRequest(workflowRequestId, ApprovalTaskServiceImpl.APPROVED);
+        }
+    }
+
+    private void validateApprovers(String taskId) throws WorkflowEngineException {
+
+        String userId = CarbonContext.getThreadLocalCarbonContext().getUserId();
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+
+        boolean isAssignedApprovalTask = false;
+        ApproverDTO approverDTO = approvalTaskDAO.getApproverDetailForApprovalTask(taskId);
+        if (ENTITY_TYPE_USERS.equals(approverDTO.getApproverType()) ||
+                ENTITY_TYPE_CLAIMED_USERS.equals(approverDTO.getApproverType())) {
+            if (approverDTO.getApproverName().equals(userId)) {
+                isAssignedApprovalTask = true;
+            }
+        } else {
+            List<String> roleIds = getAssignedRoleIds(userId, tenantDomain);
```

Comment:
rename the method
`userAssignedRoleIds`

update get the method name of approverDTO. ex:    getApproverId()